### PR TITLE
Make PolicyResult factory methods public

### PR DESCRIPTION
**Pull request by jiimaho**
In UnitTests we must be able to create fake results. Please review the comments.

### CODE REVIEW

**Comment by reisenberger** [comment]
@jiimaho Thanks!  LGTM re `internal` -> `static` changes.  I can review comments and dress up as a new version number for nuget release.

Looks like we need to [do some work on the build](https://ci.appveyor.com/project/joelhulen/polly/build/1.0.407) tho - might be few days before I can get to that.  Looks like [cake v0.22 intentionally breaks cake plug-ins](https://cakebuild.net/blog/2017/09/cake-v0.22.0-released), and we either need to [pin our version of cake core](https://cakebuild.net/docs/tutorials/pinning-cake-version) (except instructions there don't match the Polly build- need to dig deeper) or wait for the [Cake.FileHelpers plugin to update - due shortly](https://github.com/cake-contrib/Cake.FileHelpers/issues/15).

**Comment by jiimaho** [comment]
Great, thanks a bunch for all your help and rapid feedback @reisenberger .

**Comment by reisenberger** [comment]
@jiimaho  It looks like the cake build plug-in has been updated (fixes our build), so I'll make some suggestions about the intellisense for the PR shortly, then we should be able to merge it!

File: src/Polly.Shared/PolicyResult.cs [hunk 1]
```diff
@@ -103,12 +116,25 @@ internal PolicyResult(TResult result, OutcomeType outcome, Exception finalExcept
         /// </summary>
         public Context Context { get; }
 
-        internal static PolicyResult<TResult> Successful(TResult result, Context context)
+        /// <summary>
+        /// Builds a successful <see cref="PolicyResult" />
+        /// </summary>
+        /// <param name="result">The result</param>
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest change to))

`The result returned by execution through the policy`

File: src/Polly.Shared/PolicyResult.cs [hunk 2]
```diff
@@ -103,12 +116,25 @@ internal PolicyResult(TResult result, OutcomeType outcome, Exception finalExcept
         /// </summary>
         public Context Context { get; }
 
-        internal static PolicyResult<TResult> Successful(TResult result, Context context)
+        /// <summary>
+        /// Builds a successful <see cref="PolicyResult" />
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest change to))

`Builds a <see cref="PolicyResult" /> representing a successful execution through the policy.`

File: src/Polly.Shared/PolicyResult.cs [hunk 3]
```diff
@@ -35,12 +35,25 @@ internal PolicyResult(OutcomeType outcome, Exception finalException, ExceptionTy
...
 
-        internal static PolicyResult Failure(Exception exception, ExceptionType exceptionType, Context context)
+
+        /// <summary>
+        /// Builds a failed <see cref="PolicyResult" />
+        /// </summary>
+        /// <param name="exception">The exception</param>
+        /// <param name="exceptionType">The exception type</param>
+        /// <param name="context">The policy context</param>
+        /// <returns></returns>
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest add, for 'returns'))

`A <see cref="PolicyResult" /> representing a failed execution through the policy.`

File: src/Polly.Shared/PolicyResult.cs [hunk 4]
```diff
@@ -103,12 +116,25 @@ internal PolicyResult(TResult result, OutcomeType outcome, Exception finalExcept
...
+        public static PolicyResult<TResult> Successful(TResult result, Context context)
         {
             return new PolicyResult<TResult>(result, OutcomeType.Successful, null, null, context);
         }
 
-        internal static PolicyResult<TResult> Failure(Exception exception, ExceptionType exceptionType, Context context)
+        /// <summary>
+        /// Builds a failed <see cref="PolicyResult" />
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest change to))

`Builds a <see cref="PolicyResult" /> representing a failed execution through the policy.`

File: src/Polly.Shared/PolicyResult.cs [hunk 5]
```diff
@@ -103,12 +116,25 @@ internal PolicyResult(TResult result, OutcomeType outcome, Exception finalExcept
...
+        /// </summary>
+        /// <param name="exception">The exception</param>
+        /// <param name="exceptionType">The exception type</param>
+        /// <param name="context">The policy context</param>
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest change to))

`The policy execution context`

File: src/Polly.Shared/PolicyResult.cs [hunk 6]
```diff
@@ -103,12 +116,25 @@ internal PolicyResult(TResult result, OutcomeType outcome, Exception finalExcept
...
+        /// <returns></returns>
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest add, for 'returns'))

`A <see cref="PolicyResult" /> representing a failed execution through the policy.`

File: src/Polly.Shared/PolicyResult.cs [hunk 7]
```diff
@@ -117,7 +143,13 @@ internal static PolicyResult<TResult> Failure(Exception exception, ExceptionType
                 context);
         }
 
-        internal static PolicyResult<TResult> Failure(TResult handledResult, Context context)
+        /// <summary>
+        /// Builds a failed <see cref="PolicyResult" />
+        /// </summary>
+        /// <param name="handledResult">The handled result</param>
+        /// <param name="context">The policy context</param>
+        /// <returns></returns>
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest add, for 'returns'))

`A <see cref="PolicyResult" /> representing a failed execution through the policy.`

File: src/Polly.Shared/PolicyResult.cs [hunk 8]
```diff
@@ -117,7 +143,13 @@ internal static PolicyResult<TResult> Failure(Exception exception, ExceptionType
                 context);
         }
 
-        internal static PolicyResult<TResult> Failure(TResult handledResult, Context context)
+        /// <summary>
+        /// Builds a failed <see cref="PolicyResult" />
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest change to))

`Builds a <see cref="PolicyResult" /> representing a failed execution through the policy.`

File: src/Polly.Shared/PolicyResult.cs [hunk 9]
```diff
@@ -117,7 +143,13 @@ internal static PolicyResult<TResult> Failure(Exception exception, ExceptionType
...
+        /// </summary>
+        /// <param name="handledResult">The handled result</param>
+        /// <param name="context">The policy context</param>
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest change to))

`The policy execution context`

File: src/Polly.Shared/PolicyResult.cs [hunk 10]
```diff
@@ -117,7 +143,13 @@ internal static PolicyResult<TResult> Failure(Exception exception, ExceptionType
                 context);
         }
 
-        internal static PolicyResult<TResult> Failure(TResult handledResult, Context context)
+        /// <summary>
+        /// Builds a failed <see cref="PolicyResult" />
+        /// </summary>
+        /// <param name="handledResult">The handled result</param>
```
Comment (reisenberger) [inline src/Polly.Shared/PolicyResult.cs]: ((suggest change to))

`The result returned by execution through the policy, which was treated as a handled failure`

**Comment by jiimaho** [comment]
Made changes to the comments and just pushed

**Comment by reisenberger** [comment]
@jiimaho Since this PR only involves changing the 5 instances of `internal` to `public` to support testing, and adding intellisense per our discussions, I am not holding up the merge for your signing of the CLA agreement (it's the kind of tiny changes the CLA bot might have deemed not worthy of the signing).  However, it would be great for any future contributions if you could:  https://cla2.dotnetfoundation.org.

Thanks for this small but important contribution!

**Pull request merged**